compiler/src/dotty/tools/dotc/transform/TransformByNameApply.scala

```diff
@@ -43,7 +43,8 @@ abstract class TransformByNameApply extends MiniPhase { thisPhase: DenotTransfor
       case formalExpr: ExprType =>
         var argType = arg.tpe.widenIfUnstable
         if (defn.isBottomType(argType)) argType = formal.widenExpr
-        def wrap(arg: Tree) = ref(defn.cbnArg).appliedToType(argType).appliedTo(arg)
+        def wrap(arg: Tree) =
+          ref(defn.cbnArg).appliedToType(argType).appliedTo(arg).withSpan(arg.span)
```

Comment:
I don't remember all the details.  I believe it was a cbn argument from another sourcefile. We do treat these situations specially when we wrap in a Hole or an Inlined. Here we have another case of such wrapping, but arguably the position does not matter, since the cbn-arg will be eliminated at erasure anyway.